### PR TITLE
Feature/logisland 472

### PR DESCRIPTION
**Pull request by mathieu-rossignol**
Implementation of InfluxDB Service and introduction of the Integration Tests Framework.

- InfluxDB service:

The main configuration requires to define the database and the list of fields vs the list of tags.
You may also set a time field with its format.
One can also define retention policy and consistency level.

- integration tests:

Developer documentation has been added for integration tests.
To run integration tests: mvn -Pintegration-tests clean verify.
A docker container with influxdb is automatically started before the tests and stopped after.
The travis configuration has been changed to now run unit tests as well as integration tests at commit time.

### CODE REVIEW

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,857 @@
...
+        {
+            runner.setProperty(service, InfluxDBControllerService.FIELDS.getName(), explicitFieldsConfigString);
+        }
+        runner.setProperty(service, InfluxDBControllerService.TIME_FIELD.getName(),
+                makeTimeFieldConfigString(measurement, timeField, format));
+        runner.setProperty(service, InfluxDBControllerService.FLUSH_INTERVAL.getName(), "1000");
+        runner.setProperty(service, InfluxDBControllerService.BATCH_SIZE.getName(), "500");
+        runner.enableControllerService(service);
+
+        runner.assertNotValid();
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java]: runner.assertNotValid(service) instead assertNotValid() test only validation of the processor

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+    @Override
+    public void init(ControllerServiceInitializationContext context) throws InitializationException {
+
+        /**
+         * Get config and establish connection to influxdb
+         */
+        super.init(context);
+
+        // Database
+        if (!context.getPropertyValue(DATABASE).isSet())
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Not needed as property is required, or should be required instead

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 3]
```diff
@@ -0,0 +1,872 @@
...
+        {
+            throw new InitializationException("Database not provided");
+        }
+        database = context.getPropertyValue(DATABASE).asString();
+
+        // Configuration mode
+        if (!context.getPropertyValue(MODE).isSet())
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Same, this should be handled by corresponding PropertyDescriptor

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 4]
```diff
@@ -0,0 +1,872 @@
...
+        if (context.getPropertyValue(RETENTION_POLICY).isSet())
+        {
+            retentionPolicy = context.getPropertyValue(RETENTION_POLICY).asString();
+        }
+
+        // Url
+        String influxUrl = context.getPropertyValue(URL).asString();
+
+        // Sanity checking
+        if (context.getPropertyValue(PASSWORD).isSet() && !context.getPropertyValue(USER).isSet())
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: This should be done in method :
protected Collection<ValidationResult> customValidate(ValidationContext context)
instead

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 5]
```diff
@@ -0,0 +1,872 @@
...
+
+        // Configuration mode
+        if (!context.getPropertyValue(MODE).isSet())
+        {
+            throw new InitializationException("Configuration mode not provided");
+        }
+        mode = CONFIG_MODE.fromValue(context.getPropertyValue(MODE).asString());
+
+        // Parse tags and fields according to defined configuration mode
+        switch (mode)
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: This should be done in method :
protected Collection customValidate(ValidationContext context)
instead. For the verify part (throwing exceptions)

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 6]
```diff
@@ -0,0 +1,872 @@
...
+
+        // Url
+        String influxUrl = context.getPropertyValue(URL).asString();
+
+        // Sanity checking
+        if (context.getPropertyValue(PASSWORD).isSet() && !context.getPropertyValue(USER).isSet())
+        {
+            throw new InitializationException("Password provided without user name");
+        }
+        if (!context.getPropertyValue(PASSWORD).isSet() && context.getPropertyValue(USER).isSet())
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: same

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java (hunk 1 above)
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java]: Done

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java (hunk 2 above)
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Property added, thx for spotting this.

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java [hunk 7]
```diff
@@ -0,0 +1,857 @@
...
+                {measurement4, measurement4Name, timeField4, format4, configMode4_0, explicitTags4_0, explicitFields4_0},
+                {measurement4, measurement4Name, timeField4, format4, configMode4_1, explicitTags4_1, explicitFields4_1},
+                {measurement4, measurement4Name, timeField4, format4, configMode4_2, explicitTags4_2, explicitFields4_2},
+                {measurement4, measurement4Name, timeField4, format4, configMode4_3, explicitTags4_3, explicitFields4_3}
+        };
+
+        return inputs;
+    }
+
+    private static void echo(String msg) {
```
[the remark this answers is not in the document]
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/integration-test/java/com/hurence/logisland/service/influxdb/InfluxDBServiceIT.java]: Done

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java (hunk 4 above)
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Done

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java (hunk 6 above)
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Done

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java (hunk 5 above)
Comment (mathieu-rossignol) [reply logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Done

**Comment by MiniPlayer** [comment]
Looks fine to me, good job !

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 8]
```diff
@@ -450,6 +399,129 @@ public void init(ControllerServiceInitializationContext context) throws Initiali
...
+        {
+            problems.add(new ValidationResult.Builder()
+                    .valid(false)
+                    .subject(this.getClass().getSimpleName())
+                    .explanation("User name provided without password.")
+                    .build());
+        }
+
+        if (!context.getPropertyValue(DATABASE).isSet())
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Not needed as property is required

File: logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java [hunk 9]
```diff
@@ -450,6 +399,129 @@ public void init(ControllerServiceInitializationContext context) throws Initiali
...
+        {
+            problems.add(new ValidationResult.Builder()
+                    .valid(false)
+                    .subject(this.getClass().getSimpleName())
+                    .explanation("Database not provided.")
+                    .build());
+        }
+
+        if (!context.getPropertyValue(MODE).isSet())
+        {
```
Comment (MiniPlayer) [inline logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java]: Not needed as property is required

**Comment by MiniPlayer** [comment]
Looks fine to me, good job !

**Pull request merged**